### PR TITLE
[timeseries] Adds timeseries examples document

### PR DESCRIPTION
**Pull request by KexinFeng**
Update the java doc to demonstrate both training and inference feature of the timeseries package.

### CODE REVIEW

File: extensions/timeseries/docs/forecast_with_M5_data.md [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+[INFO ] - Training on: cpu().
+[INFO ] - Load MXNet Engine Version 1.9.0 in 0.086 ms.
+Training:     30% |=============                           | RMSSE: 1.43, Loss: 2.42, speed: 443.11 items/sec
+```
+
+The **java source code** of this training example is [TrainTimeSeries.java](https://github.com/deepjavalibrary/djl/blob/master/examples/src/main/java/ai/djl/examples/training/TrainTimeSeries.java).
+For your reference, the counterpart **python source code** for training is also available in [5torch.py](https://gist.github.com/Carkham/a5162c9298bc51fec648a458a3437008#file-m5torch-py)
+
+## Summary
+In this blogpost, we have shown the DJL timeseries package with [M5 Forecasting data](https://github.com/deepjavalibrary/djl/blob/master/examples/src/main/java/ai/djl/examples/inference/timeseries/M5ForecastingDeepAR.java). We focuse on the inference feature, but the training feature is also available [here](https://github.com/deepjavalibrary/djl/blob/master/examples/src/main/java/ai/djl/examples/training/TrainTimeSeries.java), where you can build your own timeseries model in DJL. With these features in DJL, now users can start mining the timeseriese data conveniently in Java.
```
Comment (frankfliu) [inline extensions/timeseries/docs/forecast_with_M5_data.md:341]:
```suggestion
In this example, we have shown the DJL timeseries package with [M5 Forecasting data](https://github.com/deepjavalibrary/djl/blob/master/examples/src/main/java/ai/djl/examples/inference/timeseries/M5ForecastingDeepAR.java). We focuse on the inference feature, but the training feature is also available [here](https://github.com/deepjavalibrary/djl/blob/master/examples/src/main/java/ai/djl/examples/training/TrainTimeSeries.java), where you can build your own timeseries model in DJL. With these features in DJL, now users can start mining the timeseriese data conveniently in Java.
```

File: extensions/timeseries/docs/forecast_with_M5_data.md [hunk 2]
```diff
@@ -0,0 +1,342 @@
+# Forecast the future in a timeseries data with Deep Java Library (DJL)
+## -- Demonstration on M5forecasting and airpassenger datasests
+Junyuan Zhang, Kexin Feng
+
+<img src="https://cdn-images-1.medium.com/max/1000/1*vfVcF-ZaC_WbTt_LWlXH0w.png" width="600" />
```
Comment (frankfliu) [inline extensions/timeseries/docs/forecast_with_M5_data.md:5]: We don't need this image here

File: extensions/timeseries/docs/forecast_with_M5_data.md [hunk 3]
```diff
@@ -0,0 +1,342 @@
...
+The package contains the following two major features.
+1. It integrates DJL with [gluonTS](https://ts.gluon.ai/stable/index.html), a powerful timeseries python package. With this feature, the pretrained models in gluonTS, either with MXNet or PyTorch, can both be directly loaded into DJL for inference and deployment in Java environment. Also take a look at the python example [m5_gluonts_template](https://github.com/awslabs/gluonts/blob/dev/examples/m5_gluonts_template.ipynb). Our convention of the parameter names are the same as theirs.
+2. It contains training features, so that users can directly build and modify timeseries deep learning models in DJL within Java envinronment.
+
+In the following, we will demonstrate these features with [M5 Forecasting](https://www.kaggle.com/c/m5-forecasting-accuracy) data. We will also use the [airpassenger](https://ts.gluon.ai/stable/index.html) data to benchmark the pretrained model loaded from gluonTS. The blog is structured as follows.
```
Comment (frankfliu) [inline extensions/timeseries/docs/forecast_with_M5_data.md:14]:
```suggestion
In the following, we will demonstrate these features with [M5 Forecasting](https://www.kaggle.com/c/m5-forecasting-accuracy) data. We will also use the [airpassenger](https://ts.gluon.ai/stable/index.html) data to benchmark the pretrained model loaded from gluonTS. The example is structured as follows.

```

File: extensions/timeseries/docs/forecast_with_M5_data.md [hunk 4]
```diff
@@ -0,0 +1,342 @@
...
+Junyuan Zhang, Kexin Feng
+
+<img src="https://cdn-images-1.medium.com/max/1000/1*vfVcF-ZaC_WbTt_LWlXH0w.png" width="600" />
+
+Time series data are commonly seen in the world. They can contain valued information that helps forecast for the future, monitor the status of a procedure and feedforward a control. Generic applications includes the following: sales forecasting, stock market analysis, yield projections, process and quality control, and many many more. See [link1](https://www.itl.nist.gov/div898/handbook/pmc/section4/pmc41.htm) and [link2](https://www.influxdata.com/time-series-forecasting-methods/#:~:text=Time%20series%20forecasting%20means%20to,on%20what%20has%20already%20happened) for further examples of timeseries data.
+
+The timeseries package introduced here belongs to a deep learning framework, DeepJavaLibrary [DJL](https://github.com/deepjavalibrary/djl). It is designed for Java developers and is compatible with the existing popular deep learning engines, like PyTorch, MXNet, and Tensorflow. This library enables users to easily train and deploy deep learning models in their Java application.
+The package contains the following two major features.
```
Comment (frankfliu) [inline extensions/timeseries/docs/forecast_with_M5_data.md:10]:
```suggestion
The package contains the following two major features.

```

File: extensions/timeseries/docs/forecast_with_M5_data.md [hunk 5]
```diff
@@ -0,0 +1,342 @@
...
+```java
+plugins {
+  id 'java'
+}
+repositories {                           
+  mavenCentral()
+}
+dependencies {
+  implementation "org.apache.logging.log4j:log4j-slf4j-impl:2.17.1"
+  implementation platform("ai.djl:bom:0.20.0")
```
Comment (frankfliu) [inline extensions/timeseries/docs/forecast_with_M5_data.md:57]:
```suggestion
  implementation platform("ai.djl:bom:0.21.0")
```

File: extensions/timeseries/docs/forecast_with_M5_data.md [hunk 6]
```diff
@@ -0,0 +1,342 @@
...
+}
+```
+
+### Define your dataset
+
+In order to realize the preprocessing of time series data, we define the `TimeSeriesData` as the input of the Translator, which is used to store the feature fields and perform corresponding transformations.
+
+So for your own dataset, you need to customize the way you get the data and put it into `TimeSeriesData` as the input to the translator. In this demo, we use `M5Dataset` which is located in `M5ForecastingDeepAR.java`.
+
+The dataset path is set in the follwing code.
```
Comment (frankfliu) [inline extensions/timeseries/docs/forecast_with_M5_data.md:71]:
```suggestion
The dataset path is set in the follwing code.

```

**Review by frankfliu — APPROVED** [review]
